src/commands/utils/udfRegistration.ts

```diff
@@ -41,3 +47,44 @@ export async function selectClassesForUdfRegistration(
 
   return selectedItems?.map((item) => item.classInfo);
 }
+export interface UdfRegistrationData {
+  /** The class to register as a UDF */
+  classInfo: JarClassInfo;
+  /** The function name to use for the UDF */
+  functionName: string;
+}
+/**
+ * Prompts the user for function names for each selected class.
+ * @param selectedClasses The classes selected for UDF registration
+ * @returns Promise that resolves to UDF registration data or undefined if cancelled
+ */
+export async function promptForFunctionNames(
+  selectedClasses: JarClassInfo[],
+): Promise<UdfRegistrationData[] | undefined> {
+  const registrations: UdfRegistrationData[] = [];
+  const functionNameRegex = /^[a-zA-Z_][a-zA-Z0-9_-]*$/;
```

Comment:
We should probably make this (or the one in [`uploadArtifactOrUDF.ts`](https://github.com/confluentinc/vscode/blob/ncothren/add-function-names/src/commands/utils/uploadArtifactOrUDF.ts#L297)) an exported constant for easier use between callers (and tests)